### ISSUE TITLE
Secure HTTP configuration for web applications (FE)

### ISSUE THREAD

**Issue by MariusJulius** [keeleinstituut/tv-tolkevarav#502]

**Comment by superspartak** [comment]
"HTTP response header"

Response headers should be set in the BE, not FE.

**Comment by thenouan** [comment]
FE is still served through HTTP, which means it sends headers with response too

**Comment by KaarelKa** [comment]
@thenouan Hey, tried some options to configure this from FE side purely, but they don't seem to work, at least not without ejecting from CRA. I understand that we could also configure these from nginx ?

**Comment by thenouan** [comment]
@KaarelKa yes, no react-app ejecting is needed. tv-web docker image contains nginx which is serving static files generated during react-app build. Dockerfile in tv-web repo contains instructions for building the image and also the nginx config. Headers can be added using add_header directive in nginx conf

https://nginx.org/en/docs/http/ngx_http_headers_module.html

**Comment by KaarelKa** [comment]
Done on infrastructure level

**Comment by plakitkelly** [comment]
![image](https://github.com/keeleinstituut/tv-tolkevarav/assets/130659408/718c6a75-14bc-4105-84ee-430b03a6bbac)

**Comment by plakitkelly** [comment]
![image](https://github.com/keeleinstituut/tv-tolkevarav/assets/130659408/2eb7f6b3-8511-490f-bc14-2a8d813bdf25)

**Comment by KaarelKa** [comment]
@thenouan 

Can you check if these are relevant ?

**Comment by MariusJulius** [comment]
@KaarelKa check the list - validate if we set all to pass, will it brake FE application or not?

**Comment by KaarelKa** [comment]
@thenouan new CSP should be:

default-src 'none';
style-src 'self' fonts.googleapis.com fonts.gstatic.com;
connect-src 'self' localhost:8000;
font-src 'self' fonts.googleapis.com fonts.gstatic.com;
script-src 'self';
manifest-src 'self';
img-src 'self' i.pinimg.com;
base-uri 'none';
form-action 'self';
frame-ancestors 'self; 


Make sure you don't change it before this task is deployed. Had to do some changes in code.
